### DISCUSSION TITLE
Massive ping users

### DISCUSSION THREAD

**Discussion by whyrmph** [Rapptz/discord.py#8115] [Q&A] [answered]
Hello, I want to make the function 
_.award (some amount of currency) (member mention)_ but
The mention of the participant must be **massive**.
So an example command is:
_.award 1500 @member1 @member2 @member3_

I tried according to the documentation to use commands.Greedy, but nothing worked for me
The code was like this:
```
@commands.command()
@commands.has_permissions(administrator=True)
async def award(self, ctx, amount: int = None, *, member: commands.Greedy[discord.Member]):
 for members in member:
  open_account(user=members)
  add_bal(members, amount)
  await ctx.send (f"{members.mention}, you have been credited {amount} bits\nAccrued: {ctx.author.mention}")
```

Please help me!
I am ready for any solution!

**Comment by rtk-rnjn** [comment, marked as answer]
@eternalyoshino you guessed right! You just need to remove `*` from the function (`award`)
```py
@commands.command()
@commands.has_permissions(administrator=True)
async def award(self, ctx, amount: int = None, member: commands.Greedy[discord.Member]):
    for members in member:
        open_account(user=members)
        add_bal(members, amount)
        await ctx.send (f"{members.mention}, you have been credited {amount} bits\nAccrued: {ctx.author.mention}")
```

---

When running the command like `award 100 @user1 @user2`, the bot tries to convert both `@user1` and `@user2` as 1 single Member. Hence, there is no user ("`@user1 @user2`") like it and therefore `member` is `[]`

**Reply by whyrmph** [reply]
Thank you! This worked for me!

**Comment by LeoCx1000** [comment]
Yeah greedy is meant to be used with other arguments after it.

`async def multi_ban(ctx, members: Greedy[Member], reason: str):`

If you don't need it you can just use args

`async def command(ctx, money: int, *members: Member):`

**Reply by trevorflahardy** [reply]
Edit:
`commands.Greedy` quite hackishly now inherits `List[T]` in 2.0, there is no need to use `commands.parameter`
![image](https://user-images.githubusercontent.com/75498301/172235751-edf19c7e-df75-49c1-a3fc-47febbd96aac.png)

Ignore the contents below:

_Optionally if you wanted to make the type checker happy :)_
```python
async def multi_ban(
    ctx: commands.Context[commands.Bot], 
    members: List[discord.Member] = commands.parameter(converter=commands.Greedy[discord.Member])
) -> None: ...
```
